pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -565,25 +565,65 @@ cdef int64_t get_conversion_factor(
         return 1
 
     if from_unit == NPY_DATETIMEUNIT.NPY_FR_W:
-        return 7 * get_conversion_factor(NPY_DATETIMEUNIT.NPY_FR_D, to_unit)
+        value = get_conversion_factor(NPY_DATETIMEUNIT.NPY_FR_D, to_unit)
+        overflow_limit = INT64_MAX // 7
+        if value > overflow_limit or value < -overflow_limit:
+            raise OverflowError("result would overflow")
+        return 7 * value
```

Comment:
can we de-dup some of this with e.g.

if ...
    value = get_conversion_factor(...
    factor = 7
elif ...
    value = get_conversion_factor(...
    factor =  24
...

overflow_limit = INT64_MAX // factor
if value > ...
    raise OverflowError(...)
return factor * value

Reply:
Yea no problem - great idea